drivers/media/platform/raspberrypi/pisp_be/pisp_be.c

```diff
@@ -312,14 +312,9 @@ static void hw_queue_job(struct pispbe_dev *pispbe,
 	write_reg(pispbe, PISP_BE_GLOBAL_BAYER_ENABLE_OFFSET, hw_enables[0]);
 	write_reg(pispbe, PISP_BE_GLOBAL_RGB_ENABLE_OFFSET, hw_enables[1]);
 
-	/*
-	 * Everything else is as supplied by the user. XXX Buffer sizes not
-	 * checked!
-	 */
-	begin =	offsetof(struct pisp_be_config, global.bayer_order) /
-								sizeof(u32);
-	end = offsetof(struct pisp_be_config, axi) / sizeof(u32);
```

Comment:
The previous version used as `begin` address the offset of `global.bayer_order`.

The `pisp_be_global_config` structure is defined as

```
struct pisp_be_global_config {
	u32 bayer_enables;
	u32 rgb_enables;
	u8 bayer_order;
	u8 pad[3];
} __attribute__((packed));
```

So the previous version was skipping the first two fields.

The new version writes from `struct pisp_be_config[0]` which instead includes the two fields. Am I reading this wrong ?

Reply:
Good catch.  It's actually a bit inconsequential.  The start of the config register range is ``global.bayer_enables``, but in the old code, we skip writing to ``global.bayer_enables`` and ``global.rgb_enables`` in the bulk programming and do it with explicit single register writes.  The reason for this is so that the driver can selectively disable blocks if there was a problem with the block config or buffers were not correctly provided.  However, this really should not happen.

I'll revert it back to the old way just to be completely safe.

Reply:
> Good catch. It's actually a bit inconsequential. The start of the config register range is `global.bayer_enables`, but in the old code, we skip writing to `global.bayer_enables` and `global.rgb_enables` in the bulk programming and do it with explicit single register writes. The reason for this is so that the driver can selectively disable blocks if there was a problem with the block config or buffers were not correctly provided. However, this really should not happen.
> 
> I'll revert it back to the old way just to be completely safe.

As long as it was intentional, it's fine.

Up to you, let me know if I should collect this last version

Reply:
I've reverted back to skipping these regs in the latest version.  Let's use that for the upstream driver as well.